pkg/app/piped/cloudprovider/kubernetes/diff.go

```diff
@@ -113,6 +139,55 @@ func (r *DiffListResult) DiffString() string {
 	return b.String()
 }
 
+func diffByCommand(old, new Manifest) ([]byte, error) {
+	oldBytes, err := old.YamlBytes()
+	if err != nil {
+		return nil, err
+	}
+
+	newBytes, err := new.YamlBytes()
+	if err != nil {
+		return nil, err
+	}
+
+	oldFile, err := os.CreateTemp("", "old")
+	if err != nil {
+		return nil, err
+	}
+	defer os.Remove(oldFile.Name())
+	if _, err := oldFile.Write(oldBytes); err != nil {
+		return nil, err
+	}
+
+	newFile, err := os.CreateTemp("", "new")
+	if err != nil {
+		return nil, err
+	}
+	defer os.Remove(newFile.Name())
+	if _, err := newFile.Write(newBytes); err != nil {
+		return nil, err
+	}
+
+	cmd := "diff"
+	data, err := exec.Command(cmd, "-u", "-N", oldFile.Name(), newFile.Name()).CombinedOutput()
+	if len(data) > 0 {
```

Comment:
`CombinedOutput` is used so `data` also contains stderr. If I understand right, any error will be ignored.

Reply:
It would be helpful if you could add a test case that an error occurs

Reply:
How about like this?

```go
	var stdout, stderr bytes.Buffer
	cmd := exec.Command(cmd, "-u", "-N", oldFile.Name(), newFile.Name())
	cmd.Stdout = &stdout
	cmd.Stderr = &stderr
	if stdout.Len() > 0 {
		// diff exits with a non-zero status when the files don't match.
		// Ignore that failure as long as we get output.
		err = nil
	}
```

Reply:
Nice. That is better. Let me apply this.

Reply:
Applied this!